web/src/javascript/app.js

```diff
@@ -3,15 +3,26 @@ import {drawGame, createGameCanvas} from "./game";
 import {gameRefresh} from "./constants";
 
 const gameCanvas = createGameCanvas();
-let currentState;
 
-(new Worker("javascript/gameWebWorker.bundle.js")).onmessage = message => currentState = message.data;
+let currentState;
+const networkWorker = new Worker("javascript/gameWebWorker.bundle.js");
+networkWorker.onmessage = message => {
+  // This is to prevent Chrome's GC from deleting the worker.
+  // It's not happening on FF, I'm pretty sure it's a V8 bug.
+  // Also, it could be caused by a mix of V8 and webpack/babel.
+  if(!networkWorker) {
+    console.error("Got GCed");
+  }
+  currentState = message.data;
+};
 
 async function updateLoop() {
+  let startTime = (new Date()).valueOf();
   if(currentState) {
     drawGame(currentState, gameCanvas);
     drawLeaderboard(currentState);
   }
+  setTimeout(updateLoop, 1000/gameRefresh - ((new Date()).valueOf()) - startTime);
```

Comment:
If we want to support IE8 : [`Math.max(0, time)`](http://stackoverflow.com/a/8431015/395386) (kidding -- I don't really mind here, interesting to know though).